### DISCUSSION TITLE
Collab conflict resolution algorithm

### DISCUSSION THREAD

**Discussion by Sahil333** [dream-num/univer#1184] [Q&A | 常见问题]
I am looking at some of the mutation commands in the sheet plugin and this doc as well https://univer.work/guides/facade/websocket/ but couldn't figure out how/what conflict resolution algorithm is being used. I was reading some articles talking about Operational Transformation or CRDTs used in collaboration application for conflict resolution. How is univer doing it?

**Comment by hexf00** [comment]
Univer Official Collaboration Service Implementation currently uses the OT algorithm to realize synergy, you can use this blog post [https://univer.work/blog/ot/](https://univer.work/blog/ot/) to understand the synergy design and implementation ideas, this article has not been translated into English, perhaps you need to prepare a translation tool.
Univer does not limit how you implement it. You can also implement it yourself.

**Reply by Sahil333** [reply]
Hi @hexf00, I read the shared article. Great article! 
I don't see the collaborative editing plugin or collaboration server in the main repo. Is it distributed separately with license? If yes, what is the licensing cost for commercial use?

**Reply by hexf00** [reply]
Hi, @Sahil333 . The business plan will be launched and announced at an appropriate time. Currently, You can experience collaborative effects here. [https://univer.work/en-us/enterprises/trial-version/](https://univer.work/en-us/enterprises/trial-version/)   It is closed source, but you are free to use it freely.
